### DISCUSSION TITLE
Self Hosted Quirrel

### DISCUSSION THREAD

**Discussion by aditsachde** [quirrel-dev/quirrel#187] [Q&A] [answered]
Hello, I want to start with saying Quirrel has been functioning pretty great and feels quite nice to use! I'm attempting to self-host Quirrel using Dokku and am documenting it [here](https://github.com/aditsachde/quirrel-dokku). 

I have a couple questions:
- Does quirrel need anything other than redis? (Volumes, postgres, etc)
- I've set `RUNNING_IN_DOCKER=TRUE` (although that is already set to true in the Dockerfile) and `REDIS_URL`, do I need to set any other env variables?
- How do I get a token? The [swagger docs](https://api.quirrel.dev/documentation/index.html) list a token endpoint, but they are not listed on the [docs](https://quirrel.sachde.xyz/documentation/index.html) for my quirrel instance.

Thanks!

**Comment by Skn0tt** [comment, marked as answer]
Hi Adit! Love that you're documenting self-hosting, would you be open to contributing the result to docs.quirrel.dev? :D

You can find a list of available environment variables here: https://github.com/quirrel-dev/quirrel/blob/main/quirrel/src/api/main.ts

For self-hosted instances, you should set `PASSPHRASES` to a `:`-separated list of allowed admin passwords. These are used to issue new tokens. If there are no admin passwords set, Quirrel is in unauthenticated mode (only suitable for dev) and thus doesn't show the token endpoint in the docs.

**Reply by aditsachde** [reply]
> would you be open to contributing the result to docs.quirrel.dev? :D

Absolutely! Once I get everything sorted out, I'll open a PR to add the non-dokku specific instructions to the quirrel docs. Thanks for the help!

**Reply by aditsachde** [reply]
Two more things, I did a cursory search through the codebase and I couldn't really see what `HOST` is used for. Should this just be set to the public quirrel URL? Also, I've just been using `ghcr.io/quirrel-dev/quirrel:main`, is there a different image that is preferred?

**Reply by Skn0tt** [reply]
`HOST` is used here: https://github.com/quirrel-dev/quirrel/blob/c24773fe2797962365d4ab942990f1146fabae00/quirrel/src/api/scheduler/index.ts#L125

I think i've added it for Heroku or so, but I can't find anything in their docs 🤔

**Reply by Skn0tt** [reply]
Also: Yes, `ghcr.io/quirrel-dev/quirrel:main` is correct. Some people recommend pinning down to a specific tag for security reasons, though.

**Reply by aditsachde** [reply]
`HOST` doesn't seem to affect anything so I've not bothered to configure it. I also pinned the image hash and am using actions to create a PR on hash change. 

I'm going to give it a couple of days and make sure everything is working correctly as I haven't actually tested the self-hosted quirrel instance yet. If everything still seems ok, I'll write up some instructions for the quirrel docs.
